### PR TITLE
docs: reference Grafana Agent Flow as an OTel Collector distribution

### PR DESCRIPTION
**Pull request by tpaschalis**
This PR adds documentation references to Grafana Agent Flow's role as an OpenTelemetry collector distribution in the main landing page as well as the root of the Flow mode docs.

### CODE REVIEW

File: docs/sources/_index.md [hunk 1]
```diff
@@ -15,10 +15,9 @@ cascade:
 
 # Grafana Agent
 
-Grafana Agent is a vendor-neutral, batteries-included telemetry collector with
-configuration inspired by [Terraform][]. It is designed to be flexible,
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
+inspired by [Terraform][]. It is designed to be flexible, performant, and
+compatible with multiple ecosystems such as Prometheus and OpenTelemetry.
```
[the remark this answers is not in the document]
Comment (tpaschalis) [reply docs/sources/_index.md:20]: I'm OK with the first suggestion. 

Let me look if there's a good place for in the root README for the second snippet as well.

File: docs/sources/_index.md [hunk 2]
```diff
@@ -15,10 +15,9 @@ cascade:
 
 # Grafana Agent
 
-Grafana Agent is a vendor-neutral, batteries-included telemetry collector with
-configuration inspired by [Terraform][]. It is designed to be flexible,
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
```
Comment (cyrille-leclerc) [inline docs/sources/_index.md:18]:
```suggestion
Grafana Agent is Grafana's distribution of the OpenTelemetry Collector with configuration
```
cc @jpkrohling

**Comment by tpaschalis** [comment]
Upon Nth reading, I don't like the repeated `capabilities` wording here, but will take a fresh look a little later on.

File: docs/sources/_index.md [hunk 3]
```diff
@@ -15,10 +15,9 @@ cascade:
...
+inspired by [Terraform][]. It is designed to be flexible, performant, and
```
Comment (cyrille-leclerc) [inline docs/sources/_index.md:19]: Should we call out as primary goal brining Prometheus native pipeline to he OTel Collector in order to notable seamless integration between the Otel and Prometheus worlds?

File: docs/sources/_index.md.t [hunk 4]
```diff
@@ -15,10 +15,9 @@ cascade:
 
 # Grafana Agent
 
-Grafana Agent is a vendor-neutral, batteries-included telemetry collector with
-configuration inspired by [Terraform][]. It is designed to be flexible,
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
```
Comment (cyrille-leclerc) [inline docs/sources/_index.md.t:18]:
```suggestion
Grafana Agent is Grafana's distribution of the OpenTelemetry Collector with configuration
```
cc @jpkrohling

File: docs/sources/flow/_index.md [hunk 5]
```diff
@@ -33,6 +33,17 @@ Components allow for reusability, composability, and focus on a single task.
 * Use expressions to bind components together to build a programmable pipeline.
 * Includes a UI for debugging the state of a pipeline.
 
+{{< param "PRODUCT_NAME" >}} is a [distribution][] of the OpenTelemetry
```
Comment (cyrille-leclerc) [inline docs/sources/flow/_index.md:36]: Ditto above "is Grafana's distribution of the OpenTelemetry Collector"

File: docs/sources/flow/_index.md [hunk 6]
```diff
@@ -33,6 +33,17 @@ Components allow for reusability, composability, and focus on a single task.
...
+Collector by Grafana Labs.
+Each distribution offers a different collection of components and capabilities.
+{{< param "PRODUCT_NAME" >}} is a distribution that includes dozens of
+OpenTelemetry-native [components][] from the OpenTelemetry project and
+introduces new capabilities such as programmable pipelines, clustering support,
```
Comment (cyrille-leclerc) [inline docs/sources/flow/_index.md:41]:
```suggestion
introduces new capabilities such as native Prometheus pipelines, code defined pipelines, clustering support,
```

**Review by cyrille-leclerc — COMMENTED** [review]
Can we review the phrasing "Grafana Agent is an OpenTelemetry Collector distribution", @jpkrohling confirmed recently it's more framed as "Grafana's distribution of the OpenTelemetry Collector".

Moreover, I feel the primary value added to call out is "native Prometheus pipelines"

File: README.md [hunk 7]
```diff
@@ -1,9 +1,8 @@
 <p align="center"><img src="docs/sources/assets/logo_and_name.png" alt="Grafana Agent logo"></p>
 
-Grafana Agent is a vendor-neutral, batteries-included telemetry collector with
-configuration inspired by [Terraform][]. It is designed to be flexible,
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
```
Comment (clayton-cornell) [inline README.md:3]:
```suggestion
Grafana Agent is Grafana's distribution of the OpenTelemetry Collector distribution with configuration
```

File: docs/sources/flow/_index.md [hunk 8]
```diff
@@ -33,6 +33,17 @@ Components allow for reusability, composability, and focus on a single task.
 * Use expressions to bind components together to build a programmable pipeline.
 * Includes a UI for debugging the state of a pipeline.
 
+{{< param "PRODUCT_NAME" >}} is a [distribution][] of the OpenTelemetry
+Collector by Grafana Labs.
+Each distribution offers a different collection of components and capabilities.
+{{< param "PRODUCT_NAME" >}} is a distribution that includes dozens of
```
Comment (clayton-cornell) [inline docs/sources/flow/_index.md:39]:
```suggestion
{{< param "PRODUCT_NAME" >}} includes dozens of
```

Is it necessary to define what the distributions do?

File: docs/sources/flow/_index.md [hunk 9]
```diff
@@ -94,5 +105,8 @@ This feature is experimental, and it doesn't support all River components.
 [Tutorials]: "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow/tutorials/"
 [Tutorials]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/send-data/agent/flow/tutorials/
 [Reference]: "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow/reference/"
-[Reference]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/send-data/agent/flow/reference/
+[Reference]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/send-data/agent/flow/reference/"
+[components]: "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow/reference/"
+[components]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/send-data/agent/flow/reference/components"
+[distribution]: "https://opentelemetry.io/ecosystem/distributions/"
```
Comment (clayton-cornell) [inline docs/sources/flow/_index.md:111]: Move this link definition out of the `docs/reference` block (and when you do, remove the double quotes around the URL). The link definition can go above or below this block... I usually put them above.

**Review by clayton-cornell — COMMENTED** [review]
Overall, the changes look good. I've added a couple of comments/suggestions

File: docs/sources/flow/_index.md (hunk 8 above)
Comment (tpaschalis) [reply docs/sources/flow/_index.md]: Not sure, we were on the fence but it might be worth to explain what makes a/our distribution different.

**Comment by tpaschalis** [comment]
cc @clayton-cornell I've split the paragraph flow/_index.md into smaller chunks and replaced some duplicate words, let me know if it sounds better now

**Comment by jpkrohling** [comment]
I agree with @cyrille-leclerc that it should be phrased as

> Grafana Agent is Grafana's distribution of the OpenTelemetry Collector

File: docs/sources/flow/_index.md [hunk 10]
```diff
@@ -33,6 +33,19 @@ Components allow for reusability, composability, and focus on a single task.
...
+
+Each distribution offers a different collection of components and capabilities.
+As a distribution, {{< param "PRODUCT_NAME" >}} includes dozens of
+OpenTelemetry-native [components][] from the OpenTelemetry project and
+introduces new features such as programmable pipelines, clustering support,
+and the ability to share pipelines around the world.
+
+In addition to being an OpenTelemetry Collector distribution,
+{{< param "PRODUCT_NAME" >}} also includes first-class support for the
+Prometheus and Loki ecosystems, enabling pipelines to mix-and-match as needed.
```
Comment (clayton-cornell) [inline docs/sources/flow/_index.md:47]:
```suggestion
Prometheus and Loki ecosystems, allowing you to mix-and-match your pipelines.
```

**Review by clayton-cornell — APPROVED** [review]
One more doc change.

The text is consistent enough and in-line with what we've been discussing. I think it'll be OK as it is. It seems to meet the requirements we've discussed.

File: README.md (hunk 7 above)
Comment (tpaschalis) [reply README.md:3]: After discussing offline, we decided to go with the original verbiage for now.

**Pull request merged**